api/tests/prost.rs

```diff
@@ -36,6 +43,99 @@ fn root_identity_examples<T: RngCore + CryptoRng>(rng: &mut T) -> Vec<RootIdenti
     ]
 }
 
+// Signed contingent input examples
+fn signed_contingent_input_examples<T: RngCore + CryptoRng>(
+    block_version: BlockVersion,
+    rng: &mut T,
+) -> Vec<SignedContingentInput> {
+    let mut result = Vec::new();
+
+    let sender = AccountKey::random(rng);
+    let recipient = AccountKey::random(rng).default_subaddress();
+    let recipient2 = AccountKey::random_with_fog(rng).default_subaddress();
+    let sender_change_dest = ChangeDestination::from(&sender);
+
+    let fpr = MockFogResolver(btreemap! {
+                        recipient2
+                .fog_report_url()
+                .unwrap()
+                .to_string()
+        =>
+            FullyValidatedFogPubkey {
+                pubkey: FromRandom::from_random(rng),
+                pubkey_expiry: 1000,
+            },
+    });
+
+    let input_credentials = get_input_credentials(
+        block_version,
+        Amount::new(200, 1.into()),
+        &sender,
+        &fpr,
+        rng,
+    );
+    let mut builder = SignedContingentInputBuilder::new(
+        block_version,
+        input_credentials,
+        vec![0u64, 1u64],
+        fpr.clone(),
+        EmptyMemoBuilder::default(),
+    );
+    builder
+        .add_output(Amount::new(400, 0.into()), &recipient, rng)
+        .unwrap();
+    result.push(builder.build(rng).unwrap());
+
+    let input_credentials = get_input_credentials(
```

Comment:
Nit: This seems like a duplicate of line 70. Do we need to define this again? Since builder is mut I think we should just be able to set it?

Reply:
the builder is consumed when we call `.build`: the `.build` function takes `self` and not `&self` or `&mut self`. So the builder is gone at this line and if we try to refer to it it's just a syntax error.